src/main/java/com/contrastsecurity/sdk/ContrastSDK.java

```diff
@@ -588,14 +610,100 @@ public Traces getTraces(String organizationId, String appId, TraceFilterForm for
+        try {
+            is = makeRequest(HttpMethod.GET, urlBuilder.getNotesByApplicationUrl(organizationId, appId, traceId, form));
+            reader = new InputStreamReader(is);
+            return this.gson.fromJson(reader, TraceNotesResponse.class);
+        } finally {
+            IOUtils.closeQuietly(is);
+            IOUtils.closeQuietly(reader);
+        }
+    }
+
+
+    /**
+     * Get the available vulnerability tags in the application whose ID is passed in.
+     *
+     * @param organizationId the ID of the organization
+     * @param appId          the ID of the application
+     * @return TagsResponse object that contains the list of Tags
+     * @throws UnauthorizedException if the Contrast account failed to authorize
+     * @throws IOException           if there was a communication problem
+     */
+    public TagsResponse getVulnTagsByApplication(String organizationId, String appId) throws IOException, UnauthorizedException {
+        InputStream is = null;
+        InputStreamReader reader = null;
+        try {
+            is = makeRequest(HttpMethod.GET, urlBuilder.getVulnTagsByApplicationUrl(organizationId, appId));
+            reader = new InputStreamReader(is);
+            return this.gson.fromJson(reader, TagsResponse.class);
+        } finally {
+            IOUtils.closeQuietly(is);
+            IOUtils.closeQuietly(reader);
+        }
+    }
+
+    /**
+     * Get the available session metadata values in the application whose ID is passed in.
+     *
+     * @param organizationId the ID of the organization
+     * @param appId          the ID of the application
+     * @param form           FilterForm query parameters
+     * @return Traces object that contains the list of Trace's
+     * @throws UnauthorizedException if the Contrast account failed to authorize
+     * @throws IOException           if there was a communication problem
+     */
+    public MetadataFilterResponse getSessionMetadataForApplication(String organizationId, String appId, TraceFilterForm form) throws IOException, UnauthorizedException {
+        InputStream is = null;
+        InputStreamReader reader = null;
+        try {
+            is = makeRequest(HttpMethod.GET, urlBuilder.getSessionMetadataForApplicationUrl(organizationId, appId, form));
+            reader = new InputStreamReader(is);
+            return this.gson.fromJson(reader, MetadataFilterResponse.class);
+        } finally {
+            IOUtils.closeQuietly(is);
+            IOUtils.closeQuietly(reader);
+        }
+    }
+
+    /**
+     * Generate an attestation report for the application whose ID is passed in.
+     * @param organizationId the ID of the organization
+     * @param appId          the ID of the application
+     * @param request
+     * @throws IOException
+     * @throws UnauthorizedException
+     */
+    public void generateAttestationReport(String organizationId, String appId, AttestationCreateRequest request)

```

Comment:
@sergeib-gmail I'm not all that familiar with this feature. How long do these usually take to generate? Seconds? Minutes?

Reply:
It depends on the # of vulns/libraries in the application.  It's usually within seconds, but can be minutes for large number of findings.  I actually am not using this function in any script that was sent to a customer, was just testing it as part of a proof of concept to see if an attestation report can be generated via API.  I can take it out/leave it in, whichever you prefer.